dart/dynamics/Entity.h

```diff
@@ -63,11 +63,23 @@ class Shape;
 /// may have different policies about how/if their reference frame or name
 /// can be changed. Use the Detachable class to create an Entity whose reference
 /// Frame can be changed arbitrarily.
-class Entity
+class Entity : public common::Subscription
 {
 public:
   friend class Frame;
 
+  enum Notification {
+
+    FRAME_CHANGE_NOTICE = 1,
```

Comment:
FRAME_CHANGED sounds good. The idea of a FRAME_CHANGING notice is interesting... I suppose some agent might want to take action if object that it's tracking is about to change its frame but hasn't done so yet. I'll think about how to incorporate that, and whether there could be any negative consequences.